tests/js/common/shell/shell-statistics.js

```diff
@@ -0,0 +1,115 @@
+/*jshint globalstrict:false, strict:false */
+/*global assertEqual, assertTrue, assertMatch, fail */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief tests for client/server side transaction invocation
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2019 ArangoDB Inc, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+/// @author Copyright 2019, ArangodDB Inc, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const jsunity = require("jsunity");
+const arangodb = require("@arangodb");
+const internal = require("internal");
+const ERRORS = arangodb.errors;
+const ArangoError = require("@arangodb").ArangoError;
+const db = arangodb.db;
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test suite
+////////////////////////////////////////////////////////////////////////////////
+
+function CommonStatisticsSuite() {
+  'use strict';
+  let c;
+  return {
+    setUp: function () {
+      c = db._create("shellCommonStatTestCollection");
+    },
+    tearDown: function () {
+      db._drop(c);
+    },
+
+    testServerStats: function () {
+      // get baseline
+      let stats1 = internal.serverStatistics();
+
+      // do stuff
+      //abort
+      try {
+        db._executeTransaction({
+          collections: {},
+          action: function () {
+            var err = new Error('test');
+            err.errorNum = 1234;
+            Object.defineProperty(err, 'name', {
+              get: function () { throw new Error('Error in getter'); }
+            });
+            throw err;
+          }
+        });
+        fail();
+      } catch (err) {
+        assertTrue(err instanceof ArangoError);
```

Comment:
I think this is an uninteresting detail for this test.
What's more important is that we know the transaction threw (and aborted).
Everything else can be validated further down using the counters.